### PR TITLE
community: add aerospike vectorstore integration

### PR DESCRIPTION
**Pull request by jdogmcsteezy**
Please let me know if you see any possible areas of improvement. I would very much appreciate your constructive criticism if time allows.

**Description:**
- Added a aerospike vector store integration that utilizes [Aerospike-Vector-Search](https://aerospike.com/products/vector-database-search-llm/) add-on.
- Added both unit tests and integration tests
- Added a docker compose file for spinning up a test environment
- Added a notebook

 **Dependencies:** any dependencies required for this change
- aerospike-vector-search

 **Twitter handle:** 
- No twitter, you can use my GitHub handle or LinkedIn if you'd like

Thanks!

### CODE REVIEW

File: libs/langchain/langchain/vectorstores/aerospike.py
Comment (hwchase17) [inline libs/langchain/langchain/vectorstores/aerospike.py:1]: this sshould NOT be re-exported inside Langchain, should be community only

Comment (jdogmcsteezy) [reply libs/langchain/langchain/vectorstores/aerospike.py:1]: Ok, sorry about that. I tried to replicate what I saw

**Review by baskaryan — COMMENTED** [review]
looks like there's some merge conflicts, otherwise lgtm! happy to help resolve if you want to give me edit access

**Comment by jdogmcsteezy** [comment]
@baskaryan, I rebased master and ran `poetry lock`

**Comment by jdogmcsteezy** [comment]
@baskaryan, I have also given you write access

**Review by baskaryan — APPROVED** [review]

File: libs/community/tests/unit_tests/vectorstores/test_aerospike.py
Comment (baskaryan) [inline libs/community/tests/unit_tests/vectorstores/test_aerospike.py:13]: fyi the aerospike-vector-search pkg is documented as supported py3.8 but given how things are typed in it (e.g. `dict[str]` style typing) it doesn't actually work with <py3.9. see example failure here: https://github.com/langchain-ai/langchain/actions/runs/9166787019/job/25202866333

Comment (jdogmcsteezy) [reply libs/community/tests/unit_tests/vectorstores/test_aerospike.py:13]: @baskaryan, thanks! I was talking to the client dev on our side and noticed your changes. We will likely update our docs instead of supporting 3.8

**Pull request merged**
